### PR TITLE
adds traceid to trace too large message

### LINKED ISSUE #884: Add traceid to "Trace too large" message
**Is your feature request related to a problem? Please describe.**
Currently it can be difficult to track down "too large" traces to help the user identify what processes are creating large traces and reduce their size.

**Describe the solution you'd like**
Add the trace id to the "trace too large" error message.

**Describe alternatives you've considered**
None.

**Additional context**
That error message is generated here:
https://github.com/grafana/tempo/blob/main/modules/ingester/trace.go#L37

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

**What this PR does**:
This PR adds trace id to `trace too large message`
**Which issue(s) this PR fixes**:
Fixes #884 

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [X]  `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: modules/ingester/trace.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (t *trace) Push(_ context.Context, trace []byte) error {
 	if t.maxBytes != 0 {
 		reqSize := len(trace)
 		if t.currentBytes+reqSize > t.maxBytes {
-			return status.Errorf(codes.FailedPrecondition, "%s max size of trace (%d) exceeded while adding %d bytes", overrides.ErrorPrefixTraceTooLarge, t.maxBytes, reqSize)
+			return status.Errorf(codes.FailedPrecondition, "%s max size of trace (%d) exceeded while adding %d bytes, the traceID is %s", overrides.ErrorPrefixTraceTooLarge, t.maxBytes, reqSize, string(t.traceID))
```
Comment (joe-elliott) [inline modules/ingester/trace.go:37]: nit: %s max size of trace (%d) exceeded while adding %d bytes to trace %s

Also please use `hex.EncodeToString()` to translate the byte slice into a hex encoded string.

**Review by joe-elliott — CHANGES_REQUESTED** [review]
Thanks for the first time contribution! There are a few small changes requested.

Also, can you add a changelog entry?

File: modules/ingester/trace.go (hunk 1 above)
Comment (mritunjaysharma394) [reply modules/ingester/trace.go:37]: > Thanks for the first time contribution! There are a few small changes requested.
> 
> Also, can you add a changelog entry?

Thank you so much for the review, I will definitely do that!

Comment (mritunjaysharma394) [reply modules/ingester/trace.go:37]: > nit: %s max size of trace (%d) exceeded while adding %d bytes to trace %s
> 
> Also please use `hex.EncodeToString()` to translate the byte slice into a hex encoded string.

Thanks a lot for the suggestion, will definitely fix it :smile:

Comment (mritunjaysharma394) [reply modules/ingester/trace.go]: Fixed :smile:

**Review by mdisibio — APPROVED** [review]

**Pull request merged**